docs/resources/dds_database_user.md

```diff
@@ -0,0 +1,113 @@
+variable "user_name" {}
+variable "user_password" {}
+variable "owned_role_name" {}
+variable "owned_role_db_name" {}
+
+resource "huaweicloud_dds_database_user" "test" {
+  instance_id = var.instance_id
+
+  name     = var.user_name
+  password = var.user_password
+  db_name  = var.db_name
+
+  roles {
+    name    = var.owned_role_name
+    db_name = var.owned_role_db_name
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the DDS instance is located.
+  Changing this parameter will create a new user.
+
+* `instance_id` - (Required, String, ForceNew) Specifies the DDS instance ID to which the user belongs.
+  Changing this parameter will create a new user.
+
+* `name` - (Required, String, ForceNew) Specifies the user name.
+  The name can contain `1` to `64` characters, only letters, digits, underscores (_), hyphens (-) and dots (.) are
+  allowed. And cannot use reserved names: **drsFull** or **drsIncremental**.
+  Changing this parameter will create a new user.
+
+* `password` - (Required, String) Specifies the user password.
+  The assword content must meet the following conditions:
+  + `8` to `32` characters long.
+  + Must contains uppercase and lowercase letters, digits, and at least one special character (`~!@#%^*-_=+?`).
+
+* `db_name` - (Required, String, ForceNew) Specifies the database name to which the user belongs.
+  The name can contain `1` to `64` characters, only letters, digits and underscores (_) are allowed.
+  Changing this parameter will create a new user.
+
+  -> After a DDS instances is created, the default database is **admin**.
+
+* `roles` - (Required, List, ForceNew) Specifies the list of roles owned by the current user.
+  The [object](#dds_database_owned_roles) structure is documented below. Changing this parameter will create a new user.
+
+<a name="dds_database_owned_roles"></a>
+The `roles` block supports:
+
+* `name` - (Required, String, ForceNew) Specifies the name of role owned by the current user.
+  The name can contain `1` to `64` characters, only letters, digits, underscores (_), hyphens (-) and dots (.) are
+  allowed. Changing this parameter will create a new user.
+
+* `db_name` - (Required, String, ForceNew) Specifies the database name to which this owned role belongs.
+  Changing this parameter will create a new user.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+* `privileges` - The list of database privileges owned by the current user.
+  The [object](#dds_database_privileges) structure is documented below.
+
+* `inherited_privileges` - The list of database privileges owned by the current user, includes all privileges
+  inherited by owned roles. The [object](#dds_database_privileges) structure is documented below.
+
+<a name="dds_database_privileges"></a>
+The `privileges` block supports:
```

Comment:
the block is used by `privileges` and `inherited_privileges`

Reply:
done.